File: apps/web/src/types/sf-chat.ts
export type SfChatStatus = {
  open: boolean;
  available_agents: number;
  message: string;
  waiting_sessions?: number;
  estimated_wait_minutes?: number | null;
};

export type SfChatSession = {
  id: string;
  status: string;
  assigned_agent_id: string | null;
  assigned_agent_name: string | null;
  created_at: string;
  updated_at: string;
  queue_message: string | null;
  bot_assistant_active?: boolean;
  wait_seconds?: number | null;
};

export type SfChatMessage = {
  id: string;
  session_id: string;
  sender_user_id?: string | null;
  sender_display_name: string;
  body: string;
  created_at: string;
  is_own: boolean;
  is_system?: boolean;
  is_bot?: boolean;
};

export type SfChatPresence = {
  is_online: boolean;
  is_sf_member: boolean;
  active_session_id: string | null;
  last_seen_at: string | null;
};

export type SfChatLogoutCheck = {
  can_logout: boolean;
  reason: string | null;
  waiting_sessions: number;
  active_sessions: number;
};

export type SfChatAgentInboxItem = {
  session: SfChatSession;
  customer_display_name: string;
  customer_email: string;
  last_message_preview: string | null;
  last_message_at: string | null;
  unread_count: number;
  customer_is_typing: boolean;
  wait_seconds?: number | null;
};

export type SfChatAgentInbox = {
  items: SfChatAgentInboxItem[];
  online: boolean;
  notification_count: number;
  waiting_sessions?: number;
  estimated_wait_minutes?: number | null;
};

File: apps/api/src/star_itsm_api/schemas/sf_chat.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class SfChatStatusRead(BaseModel):
    open: bool
    available_agents: int
    message: str


class SfChatPresenceRead(BaseModel):
    is_online: bool
    is_sf_member: bool
    active_session_id: uuid.UUID | None = None
    last_seen_at: datetime | None = None


class SfChatPresenceUpdate(BaseModel):
    online: bool
    force: bool = False


class SfChatLogoutCheckRead(BaseModel):
    can_logout: bool
    reason: str | None = None
    waiting_sessions: int = 0
    active_sessions: int = 0


class SfChatMessageRead(BaseModel):
    id: uuid.UUID
    session_id: uuid.UUID
    sender_user_id: uuid.UUID | None = None
    sender_display_name: str
    body: str
    created_at: datetime
    is_own: bool = False
    is_system: bool = False


class SfChatSessionRead(BaseModel):
    id: uuid.UUID
    status: str
    assigned_agent_id: uuid.UUID | None = None
    assigned_agent_name: str | None = None
    created_at: datetime
    updated_at: datetime
    queue_message: str | None = None


class SfChatSessionCreateResponse(BaseModel):
    session: SfChatSessionRead
    messages: list[SfChatMessageRead] = Field(default_factory=list)


class SfChatMessageCreate(BaseModel):
    body: str = Field(min_length=1, max_length=4000)


class SfChatAgentInboxItem(BaseModel):
    session: SfChatSessionRead
    customer_display_name: str
    customer_email: str
    last_message_preview: str | None = None
    last_message_at: datetime | None = None
    unread_count: int = 0
    customer_is_typing: bool = False


class SfChatAgentInboxRead(BaseModel):
    items: list[SfChatAgentInboxItem]
    online: bool
    notification_count: int


class SfChatPollRead(BaseModel):
    session: SfChatSessionRead | None = None
    messages: list[SfChatMessageRead] = Field(default_factory=list)
    status: SfChatStatusRead


class SfChatCreateTicketBody(BaseModel):
    title: str | None = Field(default=None, min_length=3, max_length=256)
